pkg/cmd/indices/config/export/export.go

```diff
@@ -0,0 +1,150 @@
+	"github.com/spf13/cobra"
+
+	indiceConfig "github.com/algolia/cli/pkg/cmd/shared/config"
+	"github.com/algolia/cli/pkg/cmdutil"
+	"github.com/algolia/cli/pkg/config"
+	"github.com/algolia/cli/pkg/iostreams"
+	"github.com/algolia/cli/pkg/utils"
+)
+
+type ExportOptions struct {
+	Config config.IConfig
+	IO     *iostreams.IOStreams
+
+	Indices   []string
+	Scope     []string
+	Directory string
+
+	SearchClient func() (*search.Client, error)
+}
+
+type ConfigJson struct {
+	Settings *search.Settings `json:"settings,omitempty"`
+	Rules    []search.Rule    `json:"rules,omitempty"`
+	Synonyms []search.Synonym `json:"synonyms,omitempty"`
+}
+
+// NewExportCmd creates and returns an export command for indices config
+func NewExportCmd(f *cmdutil.Factory, runF func(*ExportOptions) error) *cobra.Command {
+	opts := &ExportOptions{
+		IO:           f.IOStreams,
+		Config:       f.Config,
+		SearchClient: f.SearchClient,
+	}
+
+	cmd := &cobra.Command{
+		Use:               "export <index>",
+		ValidArgsFunction: cmdutil.IndexNames(opts.SearchClient),
+		Short:             "Export the config of one or multiple index(es)",
+		Long: heredoc.Doc(`
+			Export the config of index(es) including settings, synonyms and rules.
+		`),
+		Example: heredoc.Doc(`
+			# Export the config of the index 'TEST_PRODUCTS' to a .json in the current folder
+			$ algolia indices config export TEST_PRODUCTS
+
+			# Export the config of indexes 'TEST_PRODUCTS_1', 'TEST_PRODUCTS_2' and 'TEST_PRODUCTS_3' to a .json in the current folder
+			$ algolia indices config export TEST_PRODUCTS_1 TEST_PRODUCTS_2 TEST_PRODUCTS_3
+
+			# Export the synonyms and rules of the index 'TEST_PRODUCTS' to a .json in the current folder
+			$ algolia indices config export TEST_PRODUCTS --scope synonyms,rules
+
+			# Export the config of the index 'TEXT_PRODUCTS' to a .json into 'exports' folder
+			$ algolia indices config export TEST_PRODUCTS --directory exports
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.Indices = args
+
+			return runExportCmd(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.Directory, "directory", "d", "", "Directory path of the output file (directy must exist)")
+	cmd.Flags().StringSliceVarP(&opts.Scope, "scope", "s", []string{"settings", "synonyms", "rules"}, "Scope to export (default: all)")
+	_ = cmd.RegisterFlagCompletionFunc("scope",
+		cmdutil.StringSliceCompletionFunc(map[string]string{
+			"settings": "copy only the settings",
+			"synonyms": "copy only the synonyms",
+			"rules":    "copy only the rules",
+		}))
+
+	return cmd
+}
+
+func runExportCmd(opts *ExportOptions) error {
+	cs := opts.IO.ColorScheme()
+	client, err := opts.SearchClient()
+	if err != nil {
+		return err
+	}
+
+	for _, indexName := range opts.Indices {
+		indice := client.InitIndex(indexName)
+		var configJson ConfigJson
+
+		if utils.Contains(opts.Scope, "synonyms") {
+			rawSynonyms, err := indiceConfig.GetSynonyms(indice)
```

Comment:
How about moving the below logic to some interface (outside of the command logic) so we will have all the config import / export logic living in the same place and this could be re-used elsewhere?

Reply:
This could live in the `shared/config/config.go` WDYT?

Reply:
I tried to do so when building the `import` but I realised there's no shared code between import and export

However, I agree, it may be better to put everything related to import/export to the `shared/config/config.go` file: we may use these functions for `config copy` in the near future 👀 

Do you mind if I do so in this PR https://github.com/algolia/cli/pull/79? it'd be easier for me to manage this